### ISSUE TITLE
azure_rm_virtualmachine enable_automatic_updates error when not specified

### ISSUE THREAD

**Issue by mdicerbo** [ansible/ansible_collections_azure#1332]
##### SUMMARY
when trying to change vm size of a windows vm with enableAutomaticUpdates set to false, resize fails with the following error (on Azure Portal)

Changing property 'windowsConfiguration.enableAutomaticUpdates' is not allowed.

property enable_automatic_updates is not specified on task

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
azure_rm_virtualmachine

##### ANSIBLE VERSION
ansible version 2.14.5

```

##### COLLECTION VERSION
azure.azcollection 1.19


##### OS / ENVIRONMENT
target OS windows server 2019


##### STEPS TO REPRODUCE
  tasks:
    - name: Change VM Size
      azure_rm_virtualmachine:
        resource_group: "{{ resource_group }}"
        name: "{{ vmname }}"
        started: true
        state: present
        allocated: true
        vm_size: "{{ size }}"
      register: changevmresult
      ##To force playbook fail enable lines 44 + 46 and disable lines 43 + 45
      ##12.05##failed_when: changevmresult.ansible_facts.azure_vm.powerstate != "running"
      failed_when: changevmresult.failed == "true"
      ##failed_when: changevmresult.ansible_facts.azure_vm.powerstate == "running"
      until: changevmresult.ansible_facts.azure_vm.powerstate == "running"
      ##until: changevmresult.ansible_facts.azure_vm.powerstate != "running"
      retries: 3
      delay: 5
      timeout: 1200

<!--- HINT: You can paste gist.github.com links for larger files -->

##### EXPECTED RESULTS
VM Change size without issues


##### ACTUAL RESULTS
The conditional check 'changevmresult.ansible_facts.azure_vm.powerstate == \"running\"' failed. The error was: error while evaluating conditional (changevmresult.ansible_facts.azure_vm.powerstate == \"running\"): 'dict object' has no attribute 'ansible_facts'. 'dict object' has no attribute 'ansible_facts'"
```

**Comment by maciejduk** [comment]
I am having the same issue with the following:

## Code
```YAML
  tasks:
    - name: "Gather info on VM and register into a variable"
      azure.azcollection.azure_rm_virtualmachine_info:
        resource_group: "{{ resource_group }}"
        name: "{{ vm_name }}"
      register: r_vm_info
      ignore_errors: true
    - name: "Apply tags to VM"
      azure.azcollection.azure_rm_virtualmachine:
        resource_group: "{{ resource_group }}"
        name: "{{ vm_name }}"
        started: "{% if r_vm_info.vms[0].power_state == 'running' %}true{% else %}false{% endif %}"
        append_tags: true
        tags:
          StartStop: "{{ StartStop }}"
          StartStop_ContactEmail: "{{ StartStop_ContactEmail }}"
          StartStop_ShutdownDays: "{{ StartStop_ShutdownDays }}"
          StartStop_StartupDays: "{{ StartStop_StartupDays }}"
          StartStop_TimeZone: "{{ StartStop_TimeZone }}"
          StartStop_ShutdownTime: "{{ StartStop_ShutdownTime }}"
          StartStop_StartupTime: "{{ StartStop_StartupTime }}"
          StartStop_SystemShutdown: "true"
```

## Execution result
```
TASK [Gather info on VM and register into a variable] **************************
ok: [localhost]

TASK [Apply tags to VM] ********************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Error creating or updating virtual machine XXXXXXXX - (PropertyChangeNotAllowed) Changing property 'windowsConfiguration.enableAutomaticUpdates' is not allowed.\\nCode: PropertyChangeNotAllowed\\nMessage: Changing property 'windowsConfiguration.enableAutomaticUpdates' is not allowed.\\nTarget: windowsConfiguration.enableAutomaticUpdates"}
```

## System Info
```
# /usr/share/ansible/collections/ansible_collections
Collection         Version
------------------ -------
ansible.windows    2.1.0  
azure.azcollection 1.18.1 
community.windows  2.0.0  
bash-4.4# ansible --version
/usr/local/lib/python3.8/site-packages/paramiko/transport.py:236: CryptographyDeprecationWarning: Blowfish has been deprecated
  "class": algorithms.Blowfish,
ansible [core 2.12.5.post0]
  config file = None
  configured module search path = ['/home/runner/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/site-packages/ansible
  ansible collection location = /home/runner/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.8.12 (default, Sep 21 2021, 00:10:52) [GCC 8.5.0 20210514 (Red Hat 8.5.0-3)]
  jinja version = 2.10.3
  libyaml = True
```

**Comment by Fred-sun** [comment]
@mdicerbo The 'enable_automatic_updates' attribute is not allowed to be updated. Is this attribute in your configuration different from the one you created earlier? Thank you!

**Comment by Fred-sun** [comment]
@mdicerbo I've tested this parameter locally, and whether the parameter is True or False, I can 'started', 'allocated' or add tags. Thank you!

**Comment by Fred-sun** [comment]
fixed by #1410

**Comment by mdicerbo** [comment]
Thank you very for your support! i will test it as soon as possible.

Il Mer 17 Gen 2024, 10:03 Fred-sun ***@***.***> ha scritto:
